R/MsBackendMzR.R

```diff
@@ -52,14 +52,110 @@ setMethod("readMsObject", signature(object = "MsBackendMzR",
-                      old <- common_path(dataStorage(object))
-                      dataStoragePaths <- dataStorage(object)
-                      normalizedDataStoragePaths <- normalizePath(
-                          dataStoragePaths, winslash = "/", mustWork = FALSE)
-                      dataStorage(object) <- sub(old, spectraPath,
-                                                 normalizedDataStoragePaths)
+                      object <- .mz_backend_mzr_update_storage_path(
+                          object, spectraPath)
                   }
               }
               validObject(object)
               object
           })
+
+.mz_backend_mzr_update_storage_path <- function(x, spectraPath = character()) {
+    if (!length(x)) return(x)
+    old <- common_path(dataStorage(x))
+    dataStoragePaths <- dataStorage(x)
+    normalizedDataStoragePaths <- normalizePath(
+        dataStoragePaths, winslash = "/", mustWork = FALSE)
+    spectraPath <- normalizePath(spectraPath, winslash = "/", mustWork = FALSE)
+    x@spectraData$dataStorage <- sub(old, spectraPath,
+                                     normalizedDataStoragePaths)
+    x
+}
+
+################################################################################
+##
+## alabaster saveObject/readObject
+##
+################################################################################
+#' @importMethodsFrom alabaster.base saveObject
+#'
+#' @importFrom alabaster.base saveObjectFile
+#'
+#' @importFrom alabaster.base altSaveObject
+#'
+#' @exportMethod saveObject
+#'
+#' @rdname AlabasterParam
+setMethod("saveObject", "MsBackendMzR", function(x, path, ...) {
+    dir.create(path = path, recursive = TRUE, showWarnings = FALSE)
+    saveObjectFile(path, "ms_backend_mz_r",
+                   list(ms_backend_mz_r =list(version = "1.0")))
+    tryCatch({
+        do.call(altSaveObject,
+                list(x = x@spectraData, path = file.path(path, "spectra_data")))
+    }, error = function(e) {
+        stop("failed to save 'spectraData' of ", class(x)[1L], "\n - ",
+             e$message, call. = FALSE)
+    })
+    tryCatch({
+        do.call(altSaveObject,
+                list(x = x@peaksVariables,
+                     path = file.path(path, "peaks_variables")))
+    }, error = function(e) {
+        stop("failed to save 'peaksVariables' of ", class(x)[1L], "\n - ",
+             e$message)
+    })
+})
+
+#' @importFrom alabaster.base registerValidateObjectFunction
+#'
+#' @noRd
+validateAlabasterMzBackendMzR <- function(path = character(),
+                                          metadata = list()) {
+    .check_directory_content(path, c("spectra_data", "peaks_variables"))
+}
+
+#' @importFrom alabaster.base altReadObject
+#'
+#' @importFrom alabaster.base readObject
+#'
+#' @importFrom alabaster.base registerReadObjectFunction
+#'
+#' @export readObject
+#'
+#' @noRd
+readAlabasterMzBackendMzR <- function(path = character(), metadata = list(),
```

Comment:
same `readAlabasterMsBackendMzR`

Reply:
same for `.mz_backend_mzr_update_storage_path` no ? or you meant to use "mz" and not "ms" ?

Reply:
true true. I'll change.